homeassistant/components/fritz/binary_sensor.py

```diff
@@ -82,13 +85,17 @@ def update(self) -> None:
         """Update data."""
         _LOGGER.debug("Updating FRITZ!Box binary sensors")
 
-        if self.entity_description.key == "is_connected":
-            self._attr_is_on = bool(self._fritzbox_tools.fritz_status.is_connected)
-        elif self.entity_description.key == "is_linked":
-            self._attr_is_on = bool(self._fritzbox_tools.fritz_status.is_linked)
-        elif self.entity_description.key == "firmware_update":
+        if self.entity_description.key == "firmware_update":
             self._attr_is_on = self._fritzbox_tools.update_available
             self._attr_extra_state_attributes = {
                 "installed_version": self._fritzbox_tools.current_firmware,
                 "latest_available_version": self._fritzbox_tools.latest_firmware,
             }
+
+        if self._fritzbox_tools.mesh_role == MeshRoles.SLAVE:
+            return
+
+        if self.entity_description.key == "is_connected":
```

Comment:
It would be nice to move the is_on cases to the entity description via eg a lambda function attribute.

Example:
https://github.com/home-assistant/core/blob/78442d82d184636fd12c4d26f4a9fc61d3655afc/homeassistant/components/airly/sensor.py#L100

Reply:
As per discord chat, those calls do I/O.

A rework to include then in the DataUpdateCoordinator is on plan.

Simone